.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1 +1,32 @@
-<!-- Please provide a clear and meaningful description to the CHANGELOG.md file if this PR contributes some significant changes -->
+<!--
+Thank you for your Pull Request. Please provide a description above and review
+the requirements below.
+
+Bug fixes and new features should include tests and possibly benchmarks.
+
+Contributors guide: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md
+-->
+
+#### What it does
+<!-- Include relevant issues and describe how they are addressed -->
+
+#### How to test
+<!-- Describe how a reviewer can verify changes within Theia repo -->
+
+#### Review Checklist
+<!-- For completed items, change [ ] to [x].  -->
+<!-- For inapplicable items, cross out with `[ ] foo` to `[x] ~~foo~~`.  -->
+
+- [ ] commits are signed-off: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md#sign-your-work
+- [ ] commit history is cleaned up of merge and fix up commits
+  - use `git pull -r` to pick up changes from the master
+- [ ] tests/benchmarks are defined for functionallity/perf optimizations and verified by a reviewer, not an author
+- [ ] [changelog](https://github.com/theia-ide/theia/blob/master/CHANGELOG.md) is updated
+- [ ] breaking changes are justified and recorded in the changelog
+- [ ] each new file has a proper copyright with the current year and an author as an entity
+- [ ] new dependendencies are justified and [verified](https://github.com/theia-ide/theia/wiki/Registering-CQs#wip---new-ecd-theia-intellectual-property-clearance-approach-experimental)
+- [ ] copied code are justified and [approved via a CQ](https://github.com/theia-ide/theia/wiki/Registering-CQs#case-3rd-party-project-code-copiedforked-from-another-project-into-eclipse-theia-maintained-by-us)
+- [ ] new code is aligned with [project organization](https://github.com/theia-ide/theia/wiki/Code-Organization) and [coding conventions](https://github.com/theia-ide/theia/wiki/Coding-Guidelines)
+- [ ] each review approve is provided with a comment regarding whether functionallity is tested, design or code are aligned with project organization and conventions
```

Comment:
The last point in the check list ("approve comment") seem to imply that the reviewer built and tested the code. If this is required, I would make it an explicit item in the check list. One could still review the code, but not approve.

Would it make sense to require instructions on how to test the changed behaviour? I figure it's not always obvious how to do that: we had some such cases implementing VS Code API.

Reply:
> The last point in the check list ("approve comment") seem to imply that the reviewer built and tested the code. If this is required, I would make it an explicit item in the check list. One could still review the code, but not approve.

sounds good 

> Would it make sense to require instructions on how to test the changed behaviour? I figure it's not always obvious how to do that: we had some such cases implementing VS Code API.

A contributor should think about it and fill in `How to test` section, no? How he can know otherwise that he is doing the right thing.

Reply:
for VS Code API one should create a vscode extension and test it against VS Code and Theia, or write a unit test against VS Code APIs. Do you mean that we should have such recommendations?